fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/FDBLuceneIndexFailureTest.java

```diff
@@ -269,34 +454,138 @@ void addDocumentWithUnknownExceptionTest(boolean useLegacyAsyncToSync) throws IO
         try (FDBRecordContext context = openContext(contextProps)) {
             schemaSetup.accept(context);
             // Inject failures (using partition 0)
-            final MockedFDBDirectory mockedDirectory = (MockedFDBDirectory)injectMockDirectoryManager(context, index).getDirectory(groupingKey, 0);
-            mockedDirectory.addFailure(MockedFDBDirectory.Methods.LIST_ALL,
-                    new IOException((new UnknownRuntimeException("Blah"))),
-                    0);
+            getMockedDirectory(groupingKey, index, 0)
+                    .addFailure(MockedFDBDirectory.Methods.LIST_ALL,
+                            new IOException((new UnknownRuntimeException("Blah"))),
+                            0);
             // this should fail with injected exception
             assertThrows(UnknownRuntimeException.class,
                     () -> recordStore.saveRecord(createComplexDocument(1000L , ENGINEER_JOKE, docGroupFieldValue, 1)));
         }
     }
 
-    private @Nonnull FDBDirectoryManager injectMockDirectoryManager(final FDBRecordContext context, final Index index) {
-        IndexMaintainerState state = new IndexMaintainerState(recordStore, index, IndexMaintenanceFilter.NORMAL);
-        FDBDirectoryManager manager = new MockedFDBDirectoryManager(state);
-        context.putInSessionIfAbsent(state.indexSubspace, manager);
-        return manager;
+    /**
+     * Simulate the case where the Exception injection is active.
+     * This test inserts an exception mapper into the store to verify that exception handling is working when unknown
+     * exceptions
+     * are thrown from the legacy async-to-sync mechanism
+     *
+     * @param useLegacyAsyncToSync whether to set the legacy-async-to-sync property
+     */
+    @ParameterizedTest
+    @MethodSource("legacySyncAndMapping")
+    void saveDocumentWithMappedInjectedException(boolean useLegacyAsyncToSync, boolean useExceptionMapping) {
```

Comment:
Are there tests of search or throwing an error during the delete half of an update? those seem like two things that would be good to have.

Reply:
Added one more test that fails fairly late in the `commit()`. It's hard to know exactly where in the commit it failed because of the chaining of Futures.